crypto/cache.go

```diff
@@ -3,36 +3,29 @@ package crypto
 import (
 	"container/list"
 	"crypto/sha256"
+	"sort"
+	"strings"
 	"sync"
 
 	"github.com/relab/hotstuff"
 	"github.com/relab/hotstuff/consensus"
 )
 
-// key is used to identify a cached signature.
-// hash should be a hash of the message hash and signature together,
-// and threshold should be true if the entry was created/verified as a valid threshold signature.
-// This is to distinguish between valid aggregated signatures and valid threshold signatures.
-type key struct {
-	hash      consensus.Hash
-	threshold bool
-}
-
 type cache struct {
-	impl        consensus.CryptoImpl
+	impl        consensus.CryptoBase
 	mut         sync.Mutex
 	capacity    int
-	entries     map[key]*list.Element
+	entries     map[string]*list.Element
 	accessOrder list.List
 }
 
-// NewCache returns a new Crypto implementation that caches the results of the operations of the given CryptoImpl
+// NewCache returns a new Crypto implementation that caches the results of the operations of the given CryptoBase
```

Comment:
NewCache returns a new Crypto instance that caches the results of the operations of the provided CryptoBase.